signal/playerctl/playerctl_lib.lua

```diff
@@ -230,102 +400,131 @@ local function player_compare_name(name_a, name_b)
-    return player_compare_name(player_a.player_name, player_b.player_name)
+local function player_compare(self, a, b)
+    local player_a = self._private.lgi_Playerctl.Player(a)
+    local player_b = self._private.lgi_Playerctl.Player(b)
+    return player_compare_name(self, player_a.player_name, player_b.player_name)
+end
+
+local function get_current_player_info(self, player)
+    local title = self._private.lgi_Playerctl.Player.get_title(player) or ""
+    local artist = self._private.lgi_Playerctl.Player.get_artist(player) or ""
+    local artUrl = self._private.lgi_Playerctl.Player.print_metadata_prop(player, "mpris:artUrl") or ""
+    local album = self._private.lgi_Playerctl.Player.get_album(player) or ""
+
+    emit_metadata_signal(self, title, artist, artUrl, player.player_name, album, false)
+    playback_status_cb(self, player, player.playback_status)
+    volume_cb(self, player, player.volume)
+    loop_status_cb(self, player, player.loop_status)
+    shuffle_cb(self, player, player.shuffle)
 end
 
-local function start_manager()
-    manager = Playerctl.PlayerManager()
-    if #priority > 0 then
-        manager:set_sort_func(player_compare)
+local function start_manager(self)
+    self._private.manager = self._private.lgi_Playerctl.PlayerManager()
+
+    if #self.priority > 0 then
+        self._private.manager:set_sort_func(function(a, b)
+            return player_compare(self, a, b)
+        end)
     end
 
     -- Timer to update track position at specified interval
-    position_timer = gears.timer {
-        timeout = interval,
-        callback = position_cb
+    self._private.position_timer = gtimer {
+        timeout = self.interval,
+        callback = function()
+            position_cb(self)
+        end,
     }
 
     -- Manage existing players on startup
-    for _, name in ipairs(manager.player_names) do
-        init_player(name)
+    for _, name in ipairs(self._private.manager.player_names) do
+        init_player(self, name)
+    end
+
+    if self._private.manager.players[1] then
+        get_current_player_info(self, self._private.manager.players[1])
     end
 
+    local _self = self
+
     -- Callback to manage new players
-    function manager:on_name_appeared(name)
-        init_player(name)
+    function self._private.manager:on_name_appeared(name)
+        init_player(_self, name)
     end
 
     -- Callback to check if all players have exited
-    function manager:on_name_vanished(name)
-        if #manager.players == 0 then
-            metadata_timer:stop()
-            position_timer:stop()
-            awesome.emit_signal("bling::playerctl::no_players")
+    function self._private.manager:on_name_vanished(name)
+        if #self._private.manager.players == 0 then
+            self._private.metadata_timer:stop()
+            self._private.position_timer:stop()
+            _self:emit_signal("no_players")
+        else
+            get_current_player_info(_self, self._private.manager.players[1])
```

Comment:
The only undesired behavior I see from this is that every time a player exits, regardless of whether is was the highest priority player or not, the current highest priority player will emit all its signals. For example, if you had spotify running and that was the highest priority player, even if a lower priority player was closed the spotify signals would still be emitted. Ideally we would be able to detect when the highest priority player changes or not emit signals if their values haven't changed in that case.

Reply:
The new parameter was supposed to help with that, but take a look at 1dfbdb0